packages/compass-import-export/src/modules/import.ts

```diff
@@ -276,44 +297,62 @@ export const startImport = () => {
         delimiter,
         ignoreBlanks,
         stopOnErrors,
+        errorLogFilePath,
         exclude,
         transform,
       }
     );
 
     const abortController = new AbortController();
     const abortSignal = abortController.signal;
-    dispatch(onStarted(abortController));
+    dispatch(
+      onStarted({
+        abortController,
+        errorLogFilePath: errorLogFilePath || '',
+      })
+    );
 
-    let promise;
+    showStartingToast({
+      cancelImport: () => dispatch(cancelImport()),
+      fileName,
+    });
 
-    // TODO: log file, but probably only useful once we have the toast (COMPASS-6564)
-    //const logPath = path.join(app.getPath('logs'), path.basename(fileName) + '.log');
-    //const output = fs.createWriteStream(logPath);
+    let promise: Promise<ImportResult>;
 
-    const errors: ErrorJSON[] = [];
     const errorCallback = (err: ErrorJSON) => {
-      errors.push(err);
+      if (errors.length < 5) {
+        // Only store the first few errors in memory.
+        // The log file tracks all of them.
+        // If we are importing a massive file with many errors we don't
+        // want to run out of memory.
+        errors.push(err);
+      }
     };
 
+    let importDone = false;
+
+    // Note: This progress callback can be called
+    // after the import has completed/aborted.
     const progressCallback = _.throttle(function ({
       docsProcessed,
       docsWritten,
-      bytesProcessed,
     }: {
       docsProcessed: number;
       docsWritten: number;
-      bytesProcessed: number;
     }) {
-      // for now, call onGuesstimatedDocsTotal() so that the existing progress bar works
-      const averageSize = bytesProcessed / docsProcessed;
-      const guessedTotal = Math.max(
-        docsProcessed,
-        Math.ceil(fileSize / averageSize)
-      );
-      dispatch(onGuesstimatedDocsTotal(guessedTotal));
+      if (importDone) {
+        // When on error occurs in importing it will end the import.
+        // Sometimes this is while the stream is waiting on an insert response
+        // from the MongoDB server. In that case we ignore the progress
+        // so that the errors aren't overridden.
```

Comment:
Just for me to understand better: does it mean that even when you cancel, we still continue to write documents? If you aborted the write, shouldn't any in-progress request fail with an error instead of finishing and emitting progress? 🤔 If import function can emit onProgress after import is done, this sounds like the flow I would expect to be handled by the import code, not by the store slice

Reply:
What was happening is the insert request is in progress when the import is aborted, when that request succeeds it is emitting on progress as more documents were written. Thinking about your suggestion though it makes sense that the store shouldn't have to manage that. I'll update it so that this can't happen and we can remove this long comment.